src/sign/mod.rs

```diff
@@ -1,10 +1,57 @@
 //! DNSSEC signing.
 //!
 //! **This module is experimental and likely to change significantly.**
+//!
+//! Signatures are at the heart of DNSSEC -- they confirm the authenticity of a
+//! DNS record served by a secure-aware name server.  But name servers are not
+//! usually creating those signatures themselves.  Within a DNS zone, it is the
+//! zone administrator's responsibility to sign zone records (when the record's
+//! time-to-live expires and/or when it changes).  Those signatures are stored
+//! as regular DNS data and automatically served by name servers.
+
 #![cfg(feature = "sign")]
 #![cfg_attr(docsrs, doc(cfg(feature = "sign")))]
 
+use crate::base::iana::SecAlg;
+
+pub mod generic;
 pub mod key;
-//pub mod openssl;
+pub mod openssl;
 pub mod records;
 pub mod ring;
+
+/// Signing DNS records.
+///
+/// Implementors of this trait own a private key and sign DNS records for a zone
+/// with that key.  Signing is a synchronous operation performed on the current
```

Comment:
I don't think it being synchronous rules out I/O, it just makes it blocking but could be done from a background thread or slow running task pool.

Reply:
This documentation is to inform async users that they can call the `sign` function without worrying about blocking the thread.  It's unclear, though -- I'll fix that.